include/cuco/static_set.cuh

```diff
@@ -0,0 +1,274 @@
+ * and `cuco::double_hashing`. (see `detail/probe_sequences.cuh`)
+ * @tparam Allocator Type of allocator used for device storage
+ * @tparam Storage Slot window storage type
+ */
+
+template <class Key,
+          class Extent             = cuco::experimental::extent<std::size_t>,
+          cuda::thread_scope Scope = cuda::thread_scope_device,
+          class KeyEqual           = thrust::equal_to<Key>,
+          class ProbingScheme      = experimental::double_hashing<1,  // CG size
+                                                             cuco::murmurhash3_32<Key>,
+                                                             cuco::murmurhash3_32<Key>>,
+          class Allocator          = cuco::cuda_allocator<std::byte>,
+          class Storage            = cuco::experimental::aow_storage<2  // Window size
+                                                          >>
+class static_set {
+  static_assert(sizeof(Key) <= 8, "Container does not support key types larger than 8 bytes.");
+
+  static_assert(
+    cuco::is_bitwise_comparable_v<Key>,
+    "Key type must have unique object representations or have been explicitly declared as safe for "
+    "bitwise comparison via specialization of cuco::is_bitwise_comparable_v<Key>.");
+
+  static_assert(
+    std::is_base_of_v<cuco::experimental::detail::probing_scheme_base<ProbingScheme::cg_size>,
+                      ProbingScheme>,
+    "ProbingScheme must be a specialization of either cuco::double_hashing or "
+    "cuco::linear_probing.");
+
+ public:
+  static constexpr auto cg_size      = ProbingScheme::cg_size;  ///< CG size used to for probing
+  static constexpr auto window_size  = Storage::window_size;    ///< Window size used to for probing
+  static constexpr auto thread_scope = Scope;                   ///< CUDA thread scope
+
+  using key_type   = Key;  ///< Key type
+  using value_type = Key;  ///< Key type
+  /// Extent type
+  using extent_type =
+    decltype(std::declval<Extent>()
+               .template valid_extent<ProbingScheme::cg_size, Storage::window_size>());
+  using size_type      = typename extent_type::value_type;  ///< Size type
+  using key_equal      = KeyEqual;                          ///< Key equality comparator type
+  using allocator_type = Allocator;                         ///< Allocator type
+  using storage_type =
+    detail::storage<Storage, value_type, extent_type, allocator_type>;  ///< Storage type
+
+  using storage_ref_type    = typename storage_type::ref_type;  ///< Window storage reference type
+  using probing_scheme_type = ProbingScheme;                    ///< Probe scheme type
+  using ref_type =
+    cuco::experimental::static_set_ref<key_type,
+                                       thread_scope,
+                                       key_equal,
+                                       probing_scheme_type,
+                                       storage_ref_type>;  ///< Container reference type
```

Comment:
```suggestion
```
This alias is no longer relevant with mixin

Reply:
Yes, the intention was to provide this typedef to easily derive the reference type, in case someone wants to use it directly, e.g., when creating a hash set from existing memory.

Reply:
I do like the suggestion. This should work with an empty pack too, so the vall should look the same

Reply:
Removing empty pack for variadic template based on offline discussion. Disabling the creation of set ref with no operators.